src/trigger.c

```diff
@@ -0,0 +1,292 @@
+
+        if (listTrigger == NULL) {
+            continue;
+        }
+
+        if (!(listTrigger->flags.flags & 1)) {
+            continue;
+        }
+
+        if (listTrigger->flags.flags & 0x10) {
+            listTrigger->flags.flags |= 2;
+            continue;
+        }
+
+        if (listTrigger->flags.flags & 0x40) {
+            if (listTrigger->params2 == collisionStatus->currentWall) {
+                func_800E06C0(1);
+            }
+            if (listTrigger->params2 == collisionStatus->pushingAgainstWall) {
+                func_800E06C0(0);
+            } else {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x80) {
+            if (listTrigger->params2 != collisionStatus->currentFloor) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x80000) {
+            if (listTrigger->params2 != collisionStatus->floorBelow) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x100) {
+            if (listTrigger->params2 == collisionStatus->currentWall) {
+                collisionStatus->touchingWallTrigger = 1;
+            }
+            if ((listTrigger->params2 != collisionStatus->unk_0A) || !phys_can_player_interact()) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x400) {
+            if (listTrigger->params2 != collisionStatus->currentWall) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x200) {
+            if (listTrigger->params2 != collisionStatus->lastTouchedFloor) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x800) {
+            if ((listTrigger->params2 != collisionStatus->currentFloor) ||
+                !(gGameStatusPtr->pressedButtons & 0x8000) || (gPlayerStatus.flags & 0x2000)) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x1000) {
+            if (listTrigger->params2 != collisionStatus->lastWallHammered) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x40000) {
+            if (listTrigger->params2 != collisionStatus->currentCeiling) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x2000) {
+            if (listTrigger->params2 != collisionStatus->unk_0C) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x4000) {
+            if (listTrigger->params2 != collisionStatus->unk_0E) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x8000) {
+            if (listTrigger->params2 != collisionStatus->unk_10) {
+                continue;
+            }
+        }
+
+        if (listTrigger->flags.flags & 0x100000) {
+            f32* floats;
```

Comment:
```suggestion
            Vec3f* floats;
```
And type `listTrigger->params2` accordingly?

Reply:
It's not a vec3f - there are 4 floats. I honestly don't know what it is -it could be a struct for all we know. So I'd prefer to leave it for now and figure it out later on